substrate/client/network/src/request_responses.rs

```diff
@@ -56,15 +56,87 @@ use libp2p::{
 
 use std::{
 	collections::{hash_map::Entry, HashMap},
-	io, iter,
+	fmt, io, iter,
 	ops::Deref,
 	pin::Pin,
 	sync::Arc,
 	task::{Context, Poll},
 	time::{Duration, Instant},
+	fmt::{Display,Formatter},
 };
 
-pub use libp2p::request_response::{Config, InboundFailure, OutboundFailure, RequestId};
+pub use libp2p::request_response::{Config, RequestId};
+
+/// Adding a custom OutboundFailure, not depending on libp2p
+#[derive(Debug, thiserror::Error)]
+pub enum OutboundFailure {
+	/// The request could not be sent because a dialing attempt failed.
+	DialFailure,
+	/// The request timed out before a response was received.
+	Timeout,
+	/// The connection closed before a response was received.
+	ConnectionClosed,
+	/// The remote supports none of the requested protocols.
+	UnsupportedProtocols,
+}
+
+impl Display for OutboundFailure{
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result{
+		match self {
+			OutboundFailure::DialFailure => write!(f, "DialFailure"),
+			OutboundFailure::Timeout => write!(f, "Timeout"),
+			OutboundFailure::ConnectionClosed => write!(f, "ConnectionClosed"),
+			OutboundFailure::UnsupportedProtocols => write!(f, "UnsupportedProtocols"),
+		}
+	}
+}
```

Comment:
It would be better to keep original libp2p error messages (see https://github.com/libp2p/rust-libp2p/blob/master/protocols/request-response/src/lib.rs#L188-L202), but put them into enum declaration with `#[error()]` macro. Like this:
```
/// The request could not be sent because a dialing attempt failed.
#[error("Failed to dial the requested peer")]
DialFailure,
```
And the same for `InboundFailure`.